File: tasks/278.py
# 278. First Bad Version

class Solution:
    def firstBadVersion(self, n: int) -> int:
        d = 0
        u = n
        m = 0
        while d <= u:
            m = ((d + u) / 2 + 0.5)
            if m == d or m == u:
                break
            if isBadVersion(m) == True:
                u = m
            else:
                d = m
        return int(m)

File: solutions/278/278.go
package _278

func isBadVersion(version int) bool {
	panic(version)
}

func firstBadVersion(n int) int {
	a := 0
	b := n
	for a != b {
		c := (b + a) / 2
		if c == a {
			break
		}

		if isBadVersion(c) {
			b = c
		} else {
			a = c
		}
	}

	return b
}
